lister.cc

```diff
@@ -69,10 +69,10 @@ future<> lister::rmdir(fs::path dir) {
         if (de.type.value() == directory_entry_type::directory) {
             return rmdir(std::move(current_entry_path));
         } else {
-            return io_check(remove_file, current_entry_path.native());
+            return remove_file(current_entry_path.native());
         }
     }).then([dir] {
         // ...then kill the directory itself
-        return io_check(remove_file, dir.native());
+        return remove_file(dir.native());
     });
 }
```

Comment:
You can remove the #include that declares io_check(). That ensures lister is consistent in never performing io_check (rather than performing it for just some operations). In this regard the patch is an improvement since it adds consistency.

In another regard it's problematic as a caller can assume that the lower level is responsible for io_check, and thus we lose the check where it's needed. Maybe we should pass an io_error_handler as a parameter to force the caller to make a decision, rather than assume something. We can add a relaxed_error_handler that logs and forwards the exception.